crates/iceberg/src/expr/visitors/expression_evaluator.rs

```diff
@@ -338,7 +338,7 @@ mod tests {
             nan_value_counts: HashMap::new(),
             lower_bounds: HashMap::new(),
             upper_bounds: HashMap::new(),
-            key_metadata: vec![],
+            key_metadata: Some(vec![]),
```

Comment:
Do we need to use `None` by default?

Reply:
They were tests so didn't matter much, but I can change all to `None`